selectors/selectors.md

```diff
@@ -110,295 +110,269 @@ These components imply or expand into the following things:
+Uses another Selector to select things recursively. It also takes a limit on how often this recursion should happen (the limit can be set to "infinity" to keep traversing until there are no matching paths left.
 
-- `*` - means any number of single characters, except `/` -- ie matches any link within a level (as in shell `*`)
-- `**` - means any decendant (as in shell `**`)
-- `?` - matches any single character
-- `{abc,def}` - means either `abc` or `def`
-- [abc] - matches one character in the bracket set.
-- [a-z] - matches one character from the (locale-dependent) range given in the bracket.
+The recursion may also be stopped by a given CID.
 
-```go
-// Binary: <selector-code><length><glob-selector-code><binary-cid><binary-path>
-//     or: <glob-selector-code><length><binary-path>
-// String: /sel/path/<multibase-encoded-cid>/a/*/**
-type GlobSelector struct {
-    Glob Path
-}
-```
 
-### Multi Selector
+### Selector Query
 
-This is a selector that combines other selectors. It is a set expression.
+The Selector Query is a list of Selectors. The first element will be a CID Selector which points to the CID your Selector is rooted at.
 
-- `union` ($$ A \cup B $$) - takes all nodes expressed by either selectors (A _or_ B)
-- `intersection` ($$ A \cap B $$) - takes all nodes expressed by both selectors (A _and_ B)
-- `difference` ($$ A - B $$) - takes all nodes expressed in A, except those expressed in B (A *except* B)
 
-```go
-// Binary: <selector-code><length><multi-selector-code><sel-sexp>
-//     or: <multi-selector-code><length><sel-sexp>
-// String: /sel/multi/(string sel sexp)
-//       no can do
-type MultiSelector struct {
-  Op   Operator
-    Sels []Selector
-}
+Use cases
+---------
 
-type Operator int
-const (
-  OpUnion        Operator = 1
-    OpIntersection Operator = 2
-    OpDifference   Operator = 3
-)
-```
-
-#### Selector S-expressions (`sel-sexp`)
+### Deeply nested path
 
-[S-expressions](https://en.wikipedia.org/wiki/S-expression) are a simple way to combine objects into nested expressions. We define this in order to have Multi Selectors. A selector s-expression (`sel-sexp`) is defined as follows.
+In some DAG you want to get one specific value you know the path of. Let's say you want to get the birth year of a specific character of a specific show.
 
-Option 1 - simpler, need to nest another multiselector to nest another expression (this seems better).
+Example data (as JSON) with CID `cidabcdef`. To keep it simple it's represented as a single JSON structure, but imagine e.g. the characters are linked by CID.
 
+```json
+{
+  "show": "start-trek-voyager",
+  "characters": {
+    "kathrin-janeway": {
+      "birthday": {
+        "day": 20,
+        "month": 5,
+        "year": 2328
+      },
+      "rank": "vice admiral"
+    }
+  }
+}
 ```
-<sel-sexp>  ::= <varint-2+><operator>(<selector>)+
+
+The final Selector could look like this:
```

Comment:
I thought selectors were normally described as strings? do you want to include the notation that a client might use to say look up the rank of Katherin?

Reply:
No, currently there is no simple string encoding, it's always a structure that can e.g. be represented as JSON. So find out her rank would be:

```json
{
  "cidRootedSelector": {
    "root": "cidabcdef",
    "selectors": [
      {"selectPath": "characters"},
      {"selectPath": "kathryn-janeway"},
      {"selectPath": "rank"}
    ]
  }
}
```

Though one future string representation will most likely look like `cidabcdef/characters/kathryn-janeway/rank`.